src/Workspaces/Core/Portable/FindSymbols/FindReferences/BaseTypeFinder.cs

```diff
@@ -2,71 +2,61 @@
 
             // The type scenario. Iterate over all base classes to find overridden and hidden (new/Shadows) methods.
             foreach (var type in FindBaseTypes(symbol.ContainingType))
             {
                 foreach (var member in type.GetMembers(symbol.Name))
                 {
-                    var sourceMember = await SymbolFinder.FindSourceDefinitionAsync(
-                        SymbolAndProjectId.Create(member, project.Id),
-                        solution,
-                        cancellationToken).ConfigureAwait(false);
+                    cancellationToken.ThrowIfCancellationRequested();
 
-                    if (sourceMember.Symbol != null)
+                    // Add to results overridden members only. Do not add hidden members.
+                    if (SymbolFinder.IsOverride(solution, symbol, member, cancellationToken))
                     {
-                        // Add to results overridden members only. Do not add hidden members.
-                        if (SymbolFinder.IsOverride(solution, symbol, sourceMember.Symbol, cancellationToken))
-                        {
-                            results.Add(sourceMember);
-                        }
+                        results.Add(member);
 
-                        // For both overridden and inherited members, 
-                        // find all explicit and implicit interface implementations.
-                        // We need to start from each base class for cases like N() Implements I.M() 
-                        // where N() can be hidden or overwritten in a nested class later on.
-                        interfaceImplementations.AddRange(member.ExplicitOrImplicitInterfaceImplementations());
+                        // We should add implementations only for overridden members but not for hidden ones.
+                        // In the following example:
+                        // interface I { void M(); }
+                        // class A : I { public void M(); }
+                        // class B : A { public new void M(); }
+                        // we should not find anything for B.M() because it does not implement the interface:
+                        // I i = new B(); i.M(); 
+                        // will call the method from A.
+                        // However, if we change the code to 
+                        // class B : A, I { public new void M(); }
+                        // then
+                        // I i = new B(); i.M(); 
+                        // will call the method from B. We should find the base for B.M in this case.
+                        // And if we change 'new' to 'override' in the original code and add 'virtual' where needed, 
+                        // we should find I.M as a base for B.M(). And the next line helps with this scenario.
+                        results.AddRange(member.ExplicitOrImplicitInterfaceImplementations());
```

Comment:
I'm worried this is actually a bug in ExplicitOrImplicitInterfaceImplementations:

https://github.com/dotnet/roslyn/blob/1de29ff0ae5661a0ac4b8208ac42570bce4c9417/src/Workspaces/Core/Portable/Shared/Extensions/ISymbolExtensions.cs#L121

It looks like this is finding B, seeing it implements I, and then asks I what the explicit implementation is in B -- is it returning A.M or nothing at all?

Reply:
I think that Explicit in this context means VB implementations and C# ones like void I.M(). And implicit are all other.

Then, in the scenario
abstract class C : I { public abstract void [|M|]() { } }
class D : C { public override void $$M() { } }
interface I { void [|M|](); }

In the line you mentioned
containingType.FindImplementationForInterfaceMember(memberInInterface))
is called for containingType = "D" and memberInInterface = "I.M()".

And it returns "C.M()" not "D.M()". Then it is rejected by symbol.Equals.

---
In reply to: [335224689](https://github.com/dotnet/roslyn/pull/39055#discussion_r335224689) [](ancestors = 335224689)